### PR TITLE
Implemented CSV translator that handles CSV input/output support for XGBoost

### PR DESCRIPTION
**Pull request by smouaa**
## Description ##

**Prev PR: https://github.com/deepjavalibrary/djl/pull/3793**

**Summary**
Adds CSV translator support to DJL, enabling direct CSV input/output processing for tabular machine learning models

**Changes**
CSV Translator Implementation: Added CsvTranslator class in basicdataset module

NoopServingTranslatorFactory Integration: Used reflection as suggested instead to load CsvTranslator for CSV input/out processing

Input Processing: Implements CSV parsing with automatic header detection, numeric validation, and consistent column count enforcement

Output Processing: Supports both CSV and JSON output formats based on Accept header specification

Build Configuration: Added commons-csv dependency and runtime integration between API and basicdataset modules

**Test Coverage:**

CSV input with JSON output

CSV input with CSV output

Header row detection and skipping

Error handling for malformed data (uneven rows, non-numeric values, empty input)

Edge cases and validation scenarios

### CODE REVIEW

**Comment by frankfliu** [comment]
Ignore my `CsvDataset` comments. I think we should match what pandas behavior, load the csv into single 2D matrix.

Please add a unit test.

File: basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    private final CSVFormat csvFormat;
+
+    /**
+     * Constructs a CsvTranslator.
+     *
+     * @param arguments the arguments (unused but required for reflection)
+     */
+    @SuppressWarnings({"PMD.UnusedFormalParameter", "deprecation"})
+    public CsvTranslator(Map<String, ?> arguments) {
+        this.csvFormat = CSVFormat.newFormat(',').withRecordSeparator("\n");
```
Comment (frankfliu) [inline basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java:45]: use CSVFormat.INFORMIX_UNLOAD_CSV?

File: basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                if (!headerSkipped && isHeaderRow(record)) {
+                    headerSkipped = true;
+                    continue;
+                }
+
+                if (expectedCols == -1) {
+                    expectedCols = record.size();
+                } else if (record.size() != expectedCols) {
+                    throw new TranslateException(
+                            String.format(
```
Comment (frankfliu) [inline basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java:66]: String.format is unnecessary and slow. just use:

"Row " + record.getRecordNumber() + " has " + record.size() + " columns, expected " + expectedCols

File: basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    /** {@inheritDoc} */
+    @Override
+    public String processOutput(TranslatorContext ctx, NDList list) throws TranslateException {
+        try (StringWriter writer = new StringWriter();
+                CSVPrinter printer = new CSVPrinter(writer, csvFormat)) {
+
+            for (NDArray array : list) {
+                float[] data =
+                        array.toType(ai.djl.ndarray.types.DataType.FLOAT32, false).toFloatArray();
```
Comment (frankfliu) [inline basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java:128]: import ai.djl.ndarray.types.DataType;

File: basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String processOutput(TranslatorContext ctx, NDList list) throws TranslateException {
+        try (StringWriter writer = new StringWriter();
+                CSVPrinter printer = new CSVPrinter(writer, csvFormat)) {
+
+            for (NDArray array : list) {
```
[the remark this answers is not in the document]
Comment (smouaa) [reply basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java:126]: Thanks so much for the input! Just to clarify on point #1—I was under the impression that XGBoost only returns numerical predictions, not the original column or feature names. Please let me know if I’ve misunderstood

File: basicdataset/build.gradle.kts [hunk 5]
```diff
@@ -8,6 +8,7 @@ plugins {
 dependencies {
     api(project(":api"))
     api(libs.apache.commons.csv)
+    compileOnly("com.github.spotbugs:spotbugs-annotations:4.7.3")
```
Comment (frankfliu) [inline basicdataset/build.gradle.kts:11]: Why we need this?

File: api/build.gradle.kts [hunk 6]
```diff
@@ -13,6 +13,8 @@ dependencies {
 
     testImplementation(project(":testing"))
     testImplementation(libs.testng)
+    runtimeOnly(project(":basicdataset"))
```
Comment (frankfliu) [inline api/build.gradle.kts:16]: API module should not depends on databaset. this will create a circular dependeny
